### ISSUE TITLE
Suppress IntelliJ Deprecated API inspections until JDK 21 migration is completed

### ISSUE THREAD

**Issue by AmitKumarDeoghoria** [checkstyle/checkstyle#17646]
From comment: https://github.com/checkstyle/checkstyle/pull/17638#issuecomment-3194590640
Currently, IntelliJ IDEA reports Deprecated API usage for:

new Locale(String, String) in Checker.java

new URL(String) in CommonUtil.java

These APIs are not deprecated in JDK 17, but IntelliJ is flagging them early because they are marked for deprecation in newer JDKs. This causes unnecessary noise in our inspections.

Proposed Action:

Suppress the inspection in IntelliJ configuration (project-level inspection profile) for now.

Keep using these constructors until we complete the migration to JDK 21.

Once migration is complete, replace them with the new recommended APIs:

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Locale.html#of(java.lang.String)
```
Locale.of(language, country) instead of new Locale(language, country)

URI.create(filename).toURL() instead of new URL(filename)
```

Why:

- Prevents false-positive errors in IntelliJ during active development on JDK 17.

- Maintains cleaner code without scattered @noinspection or @SuppressWarnings.

- Provides a clear plan to fix once JDK 21 is fully adopted.

**Comment by giruuuuj** [comment]
# Issue #17646: Suppress IntelliJ Deprecated API inspections until JDK 21 migration

## Summary
This PR suppresses the IntelliJ `Deprecation` inspection to prevent false-positive warnings for APIs that are deprecated in newer JDK versions but still valid in JDK 17, which is the current target for this project.

## Problem
IntelliJ IDEA was flagging the following APIs as deprecated:
- `new Locale(String, String)` in Checker.java
- `new URL(String)` in CommonUtil.java

These APIs are **not deprecated in JDK 17** but IntelliJ detects them as deprecated because they're marked for deprecation in JDK 21+. This creates unnecessary noise during active development.

## Solution
- Disable the `Deprecation` inspection in [config/intellij-idea-inspections.xml](cci:7://file:///d:/project%203/checkstyle/config/intellij-idea-inspections.xml:0:0-0:0)
- Set both main inspection and Production scope to `enabled="false"`
- This provides a clean, project-wide solution without scattered `@SuppressWarnings` annotations

## Changes Made
```xml
<!-- Before -->
<inspection_tool class="Deprecation" enabled="true" level="ERROR" enabled_by_default="false">
  <scope name="Production" level="ERROR" enabled="true"/>
</inspection_tool>

<!-- After -->
<inspection_tool class="Deprecation" enabled="false" level="ERROR" enabled_by_default="false">
  <scope name="Production" level="ERROR" enabled="false"/>
</inspection_tool>  ,, contact =Abhijeetkale605@gmail.com







---

## Next Steps

1. **Create the PR** using the instructions above
2. **Request Review** from maintainers (@romani, @rnveach, @pbludov, @strkkk)
3. **Monitor CI** to ensure no build failures
4. **Address Feedback** if any reviewers suggest changes

The branch has been successfully pushed to GitHub (`add-compact-test-reproduce-npe`) and is ready for PR creation. The commit message references the issue number and provides clear context for the changes made.

**Comment by giruuuuj** [comment]
---

## Next Steps

1. **Create the PR** using the instructions above
2. **Request Review** from maintainers (@romani, @rnveach, @pbludov, @strkkk)
3. **Monitor CI** to ensure no build failures
4. **Address Feedback** if any reviewers suggest changes

The branch has been successfully pushed to GitHub (`add-compact-test-reproduce-npe`) and is ready for PR creation. The commit message references the issue number and provides clear context for the changes made.

**Comment by romani** [comment]
we are on jdk21, so we are ready to activate Idea inspection and fix all what is required.

**Comment by ZaheerAhmadDev** [comment]
@romani ,
After reviewing the source files and inspection config, this issue appears to be fully resolved:

`Locale.of(language, country) is already in place in Checker.java`
`URI.create(filename).toURL() is already in place in CommonUtil.java`
`The Deprecation inspection is re-enabled in intellij-idea-inspections.xml`
ref for both : https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java
Is this issue ready to be closed?

**Comment by romani** [comment]
suppression from https://github.com/checkstyle/checkstyle/pull/17638/changes  are removed, issue is resovled
